internal/controller/vrg_volrep.go

```diff
@@ -113,13 +113,24 @@ func (v *VRGInstance) reconcileVolRepsAsPrimary() {
 }
 
 // reconcileVolRepsAsSecondary reconciles VolumeReplication resources for the VRG as secondary
+//
+//nolint:gocognit,cyclop,funlen
 func (v *VRGInstance) reconcileVolRepsAsSecondary() bool {
 	requeue := false
 
 	for idx := range v.volRepPVCs {
 		pvc := &v.volRepPVCs[idx]
 		log := logWithPvcName(v.log, pvc)
 
+		// Potentially for PVCs that are not deleted, e.g Failover of STS without required auto delete options
+		if !containsString(pvc.Finalizers, PvcVRFinalizerProtected) {
+			log.Info("pvc does not contain VR protection finalizer. Skipping it")
+
+			v.pvcStatusDeleteIfPresent(pvc.Namespace, pvc.Name, log)
```

Comment:
Isn't this call too late for the `protectedPVC` to be deleted? You have it on line 177, which **should** cover I; unless we successfully remove the finalizer from the PVC but fail to update the VRG. In that scenario, the PVC will be gone, and we would never reach this point because `volRepPVCs` most likely wouldn’t have it.

Reply:
True on both comments, I added it in 2 places to be sure, but the PVC disappears quite fast, and also VRG updates are not a gurantee. Saw this with env tests as well in one case.

Overall we need a better place the garbage collect the protected PVC. Currently as it exists it is not added(appended) again etc. but in cases where a PVC is deleted (which is not yet supported) we would need to remove it from the ProtectedPVCs as well.

IOW, this status delete may work at times, but does not impact the functionality at present. Planning to deal with this subsequently.